model/riscv_sys_regs.sail

```diff
@@ -567,6 +567,26 @@ function retire_instruction() -> unit = {
        then minstret = minstret + 1
 }
 
+bitfield MEnvCfg : bits(64) = {
+  HADE : 61
+  /* Fill in other fields when adding support for them */
+}
+register menvcfg : MEnvCfg
+
+function legalize_menvcfg(c : MEnvCfg, high : bool, value : xlenbits) -> MEnvCfg = {
+  if high == false & sizeof(xlen) == 32 then
+    c
+  else {
+    // If 32bit, then high is true
+    let full : bits(64) = match (sizeof(xlen)) {
+      64 => value,
+      32 => value @ 0x0000_0000
+    };
+    let c = update_HADE(c, [full[61]]);
+    c
```

Comment:
I think this `let c = ... in c` could just be `update_HADE(c, [full[61])`

Reply:
I tried this, but got an error. So, leaving the `let c =` in

Reply:
Please show the error; it should work just fine

Reply:
```
Type error:
model/riscv_sys_regs.sail:585.4-30:
585 |    update_HADE(c, [full[61]]);
    |    ^------------------------^
    | No overloading for update_HADE, tried:
    | * _update_MEnvCfg_HADE
    |    MEnvCfg is not a subtype of unit
```

Any help appreciated!

Reply:
You need to remove the semicolon at the end because you're returning the value from `update_HADE`.

Reply:
Did you try:
```
update_HADE(c, [full[61]]);
c
```
Because that would result in the error you posted, what I meant was:
```
update_HADE(c, [full[61]])
```
As the last expression in the block

Reply:
That did it! Seems obvious now, sorry. Will post a new patch set later, just wanted to quickly share that this did work as you suggested. Thanks!